### ISSUE TITLE
Issue SaveReportAsPDF Event not accesable because of file save to Path

### ISSUE THREAD

**Issue by Spycoclown** [microsoft/ALAppExtensions#12064]
Hello,

in the table "Report Selections" there is a function that is used I the cloud but it uses functionality that blocks the use of the events.

```
    local procedure SaveReportAsPDF(ReportID: Integer; RecordVariant: Variant; LayoutCode: Code[20]; ReportUsage: Enum "Report Selection Usage") FilePath: Text[250]
    var
        ReportLayoutSelectionLocal: Record "Report Layout Selection";
        FileMgt: Codeunit "File Management";
        TempBlob: Codeunit "Temp Blob";
        IsHandled: Boolean;
    begin
        OnBeforeSetReportLayout(RecordVariant, ReportUsage.AsInteger());
        FilePath := CopyStr(FileMgt.ServerTempFileName('pdf'), 1, 250);

        ReportLayoutSelectionLocal.SetTempLayoutSelected(LayoutCode);
        OnBeforeSaveReportAsPDF(ReportID, RecordVariant, LayoutCode, IsHandled, FilePath, ReportUsage, false, TempBlob);
        if not IsHandled then
            Report.SaveAsPdf(ReportID, FilePath, RecordVariant);
        OnAfterSaveReportAsPDF(ReportID, RecordVariant, LayoutCode, FilePath, false, TempBlob);

        ReportLayoutSelectionLocal.SetTempLayoutSelected('');

        Commit();
    end;
```

I would like to use this event: OnAfterSaveReportAsPDF but because the file is saved on the service tier I can't access this file, I need it in the "Temp Blob" codeunit. In the Scope of "OnPrem" it workes well but in the cloud, this event is fired but I can't access the file.

This event is used in all the actions to attach a file to a document.
![image](https://user-images.githubusercontent.com/7090519/115033642-f707d480-9eca-11eb-9b15-92847575e783.png)

It would be great if you could add to this function the file at the given "FilePath" also into the "TempBlob" then it would work also in the cloud.
Thanks

**Comment by Spycoclown** [comment]
I don't think it is extensibility-enhancement it is a bug, the function Report.SaveAsPdf should not be used anymore, why not use the new way Report.SaveAs(...) because in the code late you always upload the file in the TempBlob codeunit.

I cant use the event because of the file path, so I can't replace the function with the "ishandeld" parameter.

**Comment by DavidFeldhoff** [comment]
One question how it's going to be solved, because I think it's maybe more an "extensibility-bug" than an "extensibility-enhancement".
What we really need is the `FileManagement.BLOBImportFromServerFile(TempBlob, FilePath);` call after the Report.SaveAsPDF, so that the TempBlob is filled in the existing `OnAfterSaveReportAsPDF`-event, so that we're able to catch up the exact same output of the report (and we indeed need the exact same output for our use case due to german laws!). 
```al
local procedure SaveReportAsPDF(ReportID: Integer; RecordVariant: Variant; LayoutCode: Code[20]; ReportUsage: Enum "Report Selection Usage") FilePath: Text[250]
var
    ReportLayoutSelectionLocal: Record "Report Layout Selection";
    FileMgt: Codeunit "File Management";
    TempBlob: Codeunit "Temp Blob";
    IsHandled: Boolean;
begin
    OnBeforeSetReportLayout(RecordVariant, ReportUsage.AsInteger());
    FilePath := CopyStr(FileMgt.ServerTempFileName('pdf'), 1, 250);

    ReportLayoutSelectionLocal.SetTempLayoutSelected(LayoutCode);
    OnBeforeSaveReportAsPDF(ReportID, RecordVariant, LayoutCode, IsHandled, FilePath, ReportUsage, false, TempBlob);
    if not IsHandled then begin
        Report.SaveAsPdf(ReportID, FilePath, RecordVariant);
        FileManagement.BLOBImportFromServerFile(TempBlob, FilePath); //add this line
    end;
    OnAfterSaveReportAsPDF(ReportID, RecordVariant, LayoutCode, FilePath, false, TempBlob);

    ReportLayoutSelectionLocal.SetTempLayoutSelected('');

    Commit();
end;
```
That would be totally enough! And I think that matches exactly the definition of "extensility bug"(="Smaller suggestions for improving extensibility") and it would unblock us directly.

In case you want to refactor all your code and remove the `Report.SaveAsPDF()` call and use instead the cloud-ready `Report.SaveAs()`. Well, then it's fine for us und I would agree with the "extensibilty-enhancement" (="Larger suggestions for improving extensibility") as there's definetely more work to do. But in that case I still ask you to first just add the single `FileManagement.BLOBImportFromServerFile(TempBlob, FilePath);`-line to unblock us fast and we don't have to wait a few CUs.

Hope you can make that happen!

Cheers,
David

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 18.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.
